lib/matplotlib/axes/_base.py

```diff
@@ -4123,9 +4118,111 @@ def twiny(self):
         return ax2
 
     def get_shared_x_axes(self):
-        'Return a copy of the shared axes Grouper object for x axes'
+        """Return a copy of the shared axes Weakset object for x axes"""
         return self._shared_x_axes
 
     def get_shared_y_axes(self):
-        'Return a copy of the shared axes Grouper object for y axes'
+        """Return a copy of the shared axes Weakset object for y axes"""
         return self._shared_y_axes
+
+    def is_sharing_x_axes(self):
+        return len(self.get_shared_x_axes()) > 1
+
+    def is_sharing_y_axes(self):
+        return len(self.get_shared_y_axes()) > 1
+
+    def _unshare_axes(self, shared_axes, parent):
+        for ax in getattr(self, shared_axes):
+            if getattr(ax, parent) is self:
+                setattr(ax, parent, None)
+
+        getattr(self, shared_axes).remove(self)
+        setattr(self, shared_axes, WeakSet([self]))
+        setattr(self, parent, None)
+
+    def unshare_x_axes(self, axes=None):
+        """
+        Unshare x axis.
+
+        Parameters
+        ----------
+        axes: Axes
+            Axes to unshare. None will unshare itself.
```

Comment:
I find this parameter confusing.  If I have a reference to the axes (`otheraxes`) I want to unshare, why am I not just calling `otherax.unshare_x_axes()`?  What is the use-case here?  I'm probably missing something

Reply:
Unshare only if there is a relation between them.

Reply:
The doctoring could/should say this.